### PR TITLE
Add user category search option (Requester, Observer, Technician)

### PR DESCRIPTION
**Pull request by RomainLvr**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !33234

## Dev request

Related ticket : 33234

The customer wants to sort the "Support > Tickets" view by user category

image

### Result : 
image

### CODE REVIEW

**Comment by Rom1-B** [comment]
Wouldn't that be possible via a `global rule`, like here?

image

**Comment by RomainLvr** [comment]
> Wouldn't that be possible via a `global rule`, like here?
> 
> image

After a few checks and tests, I've deduced that it's not possible to use the global rules because you'd have to add an itemtype pointing to the ticket actors (`Ticket_User`) in the profile configuration:

image

However, this class is not compatible with the operation of global rules. Using the `User` class doesn't work either because in tickets, the relationship is made through `CommonITILActors`. If you have an idea of how to make the link, I'd love to hear from you.

**Review by AdrienClairembault — CHANGES_REQUESTED** [review]
A small PHP unit test demonstrating the feature would be nice :)

**Review by cedric-anne — APPROVED** [review]

File: src/CommonITILObject.php [hunk 1]
```diff
@@ -4800,6 +4824,30 @@ public function getSearchOptionsActors()
             ]
         ];
 
+        $tab[]= [
```
Comment (cedric-anne) [inline src/CommonITILObject.php:4827]:
```suggestion
        $tab[] = [
```

File: src/CommonITILObject.php [hunk 2]
```diff
@@ -4864,6 +4912,30 @@ public function getSearchOptionsActors()
...
```
Comment (cedric-anne) [inline src/CommonITILObject.php:4915]:
```suggestion
        $tab[] = [
```

**Review by AdrienClairembault — APPROVED** [review]
Ok for the requested changes (but still not tested)

**Review by Rom1-B — CHANGES_REQUESTED** [review]
Seems OK, but I agree with @AdrienClairembault that tests would be welcome.

**Pull request merged**